### PR TITLE
Add `README.md` for `xilem`.

### PR DESCRIPTION
**Pull request by xStrom**
This is a basic skeleton based on the latest Linebender standards.

I also added a paragraph about the reactive layer's origins.
Plus a section describing the modularity of our stack.

Additional sections describing more of the actual new state of things are welcome in follow-up PRs, but are not a goal of this PR.

### CODE REVIEW

**Review by waywardmonkeys — APPROVED** [review]

File: crates/xilem_masonry/README.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+</div>
+
+Xilem is a UI toolkit.
+It combines ideas from Flutter, SwiftUI, and Elm.
+Like all of these, it uses lightweight view objects, diffing them to provide minimal updates to a retained UI.
+Like SwiftUI, it is strongly typed.
+For more details on Xilem's reactive architecture see [Xilem: an architecture for UI in Rust].
+
+Xilem's reactive layer is built on top of a wide array of foundational Rust UI projects, e.g.:
+* Widgets are provided by [Masonry], which is a fork of [Druid].
```
[the remark this answers is not in the document]
Comment (jaredoconnell) [reply crates/xilem_masonry/README.md:24]: I see what you mean. Perhaps we can call it an official fork? Or would that be inaccurate?

**Review by jaredoconnell — APPROVED** [review]

File: crates/xilem_masonry/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (xStrom) [reply crates/xilem_masonry/README.md]: I changed it to *which is a fork of the now discontinued Druid]UI toolkit* to make it seem less hostile.

File: crates/xilem_masonry/README.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+[![Dependency staleness status.](https://deps.rs/crate/xilem/latest/status.svg)](https://deps.rs/crate/xilem)
+
+</div>
+
+Xilem is a UI toolkit.
+It combines ideas from Flutter, SwiftUI, and Elm.
+Like all of these, it uses lightweight view objects, diffing them to provide minimal updates to a retained UI.
+Like SwiftUI, it is strongly typed.
+For more details on Xilem's reactive architecture see [Xilem: an architecture for UI in Rust].
+
```
[the remark this answers is not in the document]
Comment (xStrom) [reply crates/xilem_masonry/README.md:22]: Added alpha warning.

File: crates/xilem_masonry/README.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+Xilem's reactive layer is built on top of a wide array of foundational Rust UI projects, e.g.:
+* Widgets are provided by [Masonry], which is a fork of [Druid].
+* Rendering is provided by [Vello], a high performance GPU compute-centric 2D renderer.
+* GPU compute infrastructure is provided by [wgpu].
+* Text support is provided by [Parley], [Fontique], [Swash], and [Skrifa]. 
+* Accessibility is provided by [AccessKit].
+* Window handling is provided by [winit].
```
[the remark this answers is not in the document]
Comment (xStrom) [reply crates/xilem_masonry/README.md:29]: Yeah, but I think it's interesting info for someone using Xilem. Anyway, we can discuss it in more depth for future revisions, I'll leave it as-is for now.

[the remark this answers is not in the document]
Comment (waywardmonkeys) [reply crates/xilem_masonry/README.md:29]: I like that it points out the ecosystem collaboration.

**Pull request merged**